### PR TITLE
implemented window-start

### LINKED ISSUE #312: Port window-start to Rust
```c
DEFUN ("window-start", Fwindow_start, Swindow_start, 0, 1, 0,
       doc: /* Return position at which display currently starts in WINDOW.
WINDOW must be a live window and defaults to the selected one.
This is updated by redisplay or by calling `set-window-start'.  */)
  (Lisp_Object window)
{
  return Fmarker_position (decode_live_window (window)->start);
}

```

`marker_position` is already in marker.rs

### PR DESCRIPTION
**Pull request by Yarikx**
closes #312

### CODE REVIEW

File: rust_src/src/windows.rs [hunk 1]
```diff
@@ -98,3 +98,11 @@ pub fn window_valid_p(object: LispObject) -> LispObject {
         |win| win.contents().is_not_nil(),
     ))
 }
+
+/// Return position at which display currently starts in WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+/// This is updated by redisplay or by calling `set-window-start'.
+#[lisp_fn]
+pub fn window_start(window: LispObject) -> LispObject {
+    marker_position(LispObject::from_raw(window.as_live_window_or_error().start))
```
Comment (birkenfeld) [inline rust_src/src/windows.rs]: We usually put a wrapper method around the raw struct members once we need them, e.g. `...as_live_window_or_error().start_marker()`.

**Comment by birkenfeld** [comment]
Thanks for the PR! I have one improvement you could make, it'll be in `LispWindowRef`.

File: rust_src/src/windows.rs (hunk 1 above)
Comment (Yarikx) [reply rust_src/src/windows.rs]: good point, fixed

File: rust_src/src/windows.rs [hunk 2]
```diff
@@ -98,3 +103,11 @@ pub fn window_valid_p(object: LispObject) -> LispObject {
         |win| win.contents().is_not_nil(),
     ))
 }
+
+/// Return position at which display currently starts in WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+/// This is updated by redisplay or by calling `set-window-start'.
+#[lisp_fn]
+pub fn window_start(window: LispObject) -> LispObject {
+    marker_position(window.as_live_window_or_error().start_marker())
```
Comment (birkenfeld) [inline rust_src/src/windows.rs]: Hmm, `nil` is allowed here, but `as_live_window_or_error` doesn't handle it; you have to handle that case here.

File: rust_src/src/windows.rs [hunk 3]
```diff
@@ -98,3 +103,11 @@ pub fn window_valid_p(object: LispObject) -> LispObject {
         |win| win.contents().is_not_nil(),
     ))
 }
+
+/// Return position at which display currently starts in WINDOW.
+/// WINDOW must be a live window and defaults to the selected one.
+/// This is updated by redisplay or by calling `set-window-start'.
+#[lisp_fn]
```
Comment (birkenfeld) [inline rust_src/src/windows.rs]: also needs a `min = "0"` here

Comment (Yarikx) [reply rust_src/src/windows.rs]: Thanks, indeed missed that one.

**Comment by birkenfeld** [comment]
Nice, thanks!

**Pull request merged**